src/sundials/sundials_linearsolver.c

```diff
@@ -35,21 +35,22 @@ static SUNProfiler getSUNProfiler(SUNLinearSolver S)
 
 SUNLinearSolver SUNLinSolNewEmpty(SUNContext sunctx)
 {
-  SUNLinearSolver     LS;
+  SUNFunctionBegin(sunctx);
+  SUNLinearSolver LS;
   SUNLinearSolver_Ops ops;
 
   /* a context is required */
-  if (sunctx == NULL) return(NULL);
+
```

Comment:
We could do something like `SUNAssert(SUNCTX_)` in the `SUNFunctionBegin` macro.

Reply:
Would that work though since `SUNAssert` need to access the context to report the error? Another case that would be difficult to catch would be if the object we're getting the context from is `NULL`.

Reply:
It would work if we add a global error handler as a fallback. Otherwise, there is not much we can really do error-handling wise if the context is NULL. Maybe we should treat these cases as catastrophic and just let whatever happens - happen?

Reply:
For constructors, we could still have `if (sunctx == NULL) return NULL;` just before `SUNFunctionBegin` so we don't create an object without a context.